### PR TITLE
Failing test for feature 4.

### PR DESCRIPTION
**Pull request by orhanna14**
Seems like it's building on the ideas from features 1-3. Wrote only the test here, basing it on behavior.

### CODE REVIEW

File: spec/game_runner_spec.rb [hunk 1]
```diff
@@ -25,8 +25,10 @@
 
         expect(stdout.string).to include(grid_template)
       end
+    end
 
-      it "prompts a user for input, returns a message when given invalid input, and quits when input is valid" do
+    context "it prompts the user for input" do
```
Comment (baccigalupi) [inline spec/game_runner_spec.rb]: I think about `context` blocks as starting with `when`. For example `when a user ...`. The `it` at the beginning of the context implies it is just an example. Where you trying to make descriptions more efficient with blocks? Or is there some real behavioral context here?

**Review by baccigalupi — CHANGES_REQUESTED** [review]
Logic for the added tests look great. Take a pass at making the contexts and descriptions more organized and readable.

**Review by baccigalupi — CHANGES_REQUESTED** [review]
I love these descriptions. The nesting of a single test example inside a context block is a bit much overhead. What if you just make really big description strings with the context and expectation you have.

When I am refactoring client code, I run into these nestings pretty often, and it slows down the changing of tests (which is needed regularly).

When contexts are really useful is in a long file like a controller spec. The controller in rails does not have just one responsibility. It has one responsibility per public action ... so nested context or describe statements help separate out the test work.

**Comment by orhanna14** [comment]
That's really good to know! It seems like it could be really easy to get carried away in nestings...

**Review by baccigalupi — APPROVED** [review]
Looks good!

**Pull request merged**